### PR TITLE
[REVIEW] fix pool initialization

### LINKED ISSUE #471: [BUG] Initializing pool_memory_resource with 90% of GPU memory as initial size results in OOM
**Describe the bug**
Trying to initialize `pool_memory_resource` with 90% of the GPU memory as the initial pool size and an unspecified maximum size fails with an OOM.  This used to work without an OOM.

**Steps/Code to reproduce bug**
On a 16GB GPU, try calling `pool_memory_resource` backed by a normal CUDA memory resource and with parameter initial_pool_size = 15250961203.  Instead of allocating the specified initial size (which would have succeeded) it ends up trying to allocate 16945512448 instead which fails.

**Expected behavior**
A clear and concise description of what you expected to happen.  Using `git bisect` it appears commit ae3a1fd531723c7aabda29858dbbf208b8ca6450 is where the behavior changed.

### PR DESCRIPTION
**Pull request by rongou**
Fixes #471 

@jrhemstad @harrism @jlowe

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: include/rmm/mr/device/pool_memory_resource.hpp [hunk 1]
```diff
@@ -177,10 +179,8 @@ class pool_memory_resource final
    */
   block_type block_from_upstream(size_t size, cudaStream_t stream)
   {
-    auto grow_size = size_to_grow(size);
-    RMM_EXPECTS(grow_size > 0, rmm::bad_alloc, "Maximum pool size exceeded");
-    void* p = upstream_mr_->allocate(grow_size, stream);
-    block_type b{reinterpret_cast<char*>(p), grow_size, true};
+    void* p = upstream_mr_->allocate(size, stream);
```
Comment (harrism) [inline include/rmm/mr/device/pool_memory_resource.hpp:182]: You need to align to min alignment.

**Comment by jrhemstad** [comment]
I've already got a PR with a fix. While writing a test that would catch this bug in the future I uncovered another potential bug, either in the `pool_memory_resource` itself or in the test infrastructure. Still triaging.

**Comment by rongou** [comment]
I just needed a quick fix to unblock myself. Feel free to close this.

**Comment by harrism** [comment]
Superceded by #473

**Pull request closed without merging**